### PR TITLE
Support mhlo.slice operation on Linalg on tensors path.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**

### CODE REVIEW

File: iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp [hunk 1]
```diff
@@ -194,15 +194,16 @@ static bool isDispatchableOp(Operation *op) {
...
+       op->getContext()->getLoadedDialect<linalg::LinalgDialect>()) &&
+      !isa<SubTensorOp>(op)) {
     return false;
   }
   return !isAlwaysClonedIntoDispatchOp(op);
 }
 
 static bool isAlwaysFusedIntoDispatchOp(Operation *op) {
-  return isDispatchableOp(op) && isa<linalg::TensorReshapeOp>(op);
+  return isDispatchableOp(op) && isa<linalg::TensorReshapeOp, SubTensorOp>(op);
```
Comment (asaadaldien) [inline iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp:206]: Just to help myself understand...
Does that means its going to be fused with its producer ? e.g what will happen in that case? 
```mlir
%0 = linalg.matmul ...
%1 = subtensor %0 ...
```
They should in the same dispatch workgroup right ?

Comment (MaheshRavishankar) [reply iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp:206]: No, we are mostly fusing with producer. So in this case if %1 is the output, then they will go into two separate dispatch regions. We can revisit if it becomes a performance issue.

Comment (asaadaldien) [reply iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp:206]: Thanks!

**Review by asaadaldien — APPROVED** [review]
LGTM! Feel free to submit after tests passes.

**Pull request merged**